### DISCUSSION TITLE
Looking for guidelines on rate limiting.

### DISCUSSION THREAD

**Discussion by sjvanrossum** [weather-gov/api#315] [Q&A]
**Please describe what you're trying to do.**
I've looked through most if not all of your documentation, but I can't seem to find any guidelines on rate limits imposed on API consumers. I'd like to know your recommendation for requests per second (max) and concurrently running requests (max).

An example I had been working on:
```
for STATION in api.weather.gov/stations?state=AL:
  get api.weather.gov/stations/STATION/observations/latest
```
Would batching make any difference in this case? Like:
```
get api.weather.gov/stations/STATION-1,...,STATION-N/observations/latest
```

**Comment by scadergit** [comment]
We're not allowed to share this information. 🙄 (Yes, we know...)

What I can say is that if you increase the limit and you get blocked, it's only for the few seconds that you exceeded the limit. You won't get perminately blocked unless we take manual action from continued abuse, and we'll attempt to reach out to you (if UA permits) before then.

**Comment by scadergit** [comment]
Oh, and you'll know you're blocked because the error returned is from our CDN and it looks nothing like any other resopnse from the API.

**Comment by MatthewFlamm** [comment]
I think a general guideline for API call frequency would be helpful, if possible.  There is some detail for alerts already in the [FAQ](https://www.weather.gov/documentation/services-web-alerts), which suggests no more often than 30 seconds.  There is this sentence "Refer to the API specification for more information on rate limits.", but I couldn't find anything for the weather endpoints.  A general order of magnitude suggestion like for alerts would be useful in my opinion.
